src/csharp/Microsoft.Spark/ML/Feature/NGram.cs

```diff
@@ -0,0 +1,114 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Spark.Interop;
+using Microsoft.Spark.Interop.Ipc;
+using Microsoft.Spark.Sql;
+
+namespace Microsoft.Spark.ML.Feature
+{
+    /// <summary>
+    /// Class <see cref="NGram"/> transformer that converts the input array of strings into an array of n-grams.
+    /// Null values in the input array are ignored.
+    /// It returns an array of n-grams where each n-gram is represented by a space-separated string of words.
+    /// </summary>
+    public class NGram : FeatureBase<NGram>, IJvmObjectReferenceProvider
+    {
+        private static readonly string s_NGramClassName =
```

Comment:
nit: let's call this variable s_nGramClassName? (to follow naming convention standards)